x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/blob/BlobStoreCacheMaintenanceService.java

```diff
@@ -78,10 +180,60 @@ public void clusterChanged(ClusterChangedEvent event) {
         }
         final ShardRouting primary = systemIndexPrimaryShard(state);
         if (primary == null || Objects.equals(state.nodes().getLocalNodeId(), primary.currentNodeId()) == false) {
```

Comment:
I wonder if we should also check that shard is active here?

Reply:
Yes we should, I thought I added this in #77686 🤔 I pushed 03d178bc7591b0d3a758eecd7f45380791faf4b0